functional/SyntheticGCWorkload/README.md

```diff
@@ -0,0 +1,433 @@
+# A Multi-Threaded Tunable Synthetic GC Workload
+ 
+## 1 - The Sessions Workload Application
+
+###  1.1 - Motivation and Approach
+
+Percolate collections and aborts are very costly occurrences in the J9 GC. Unfortunately they are sometimes seen on production systems running J9 and may lead to significant performance issues.
+
+Due to the inherent difficulties in reproducing the real-world conditions under which these issues often occur, they can be very difficult to debug. It is therefore desirable to find an alternate means by which we can replicate similar conditions in a controlled test environment.
+
+A synthetic workload was therefore devised in order to facilitate the debugging of these problems and the formulation of solutions. In order to find a simple set of parameters which can reproduce a given issue, a synthetic workload was devised, allowing the developer to specify a allocation and reference change patterns.
+
+The results can then be monitored using an included verbose GC parser which monitors the effects of the workload on the garbage collector. Comparing these results to results from the customer’s system allows
+the development of a workload that closely replicates the customer’s problem and allows the development of a targeted and tested solution.
+
+### 1.2 - Design
+
+The Synthetic GC Workload provides a framework for the construction of custom allocation and reference change scenarios, specified by an XML configuration file. Each configuration specifies one or more workloads which run simultaneously. Each workload has a hard runtime, after which it will terminate. Repetition of workloads and payloads is also supported.
+
+Payloads are specified within each workload, each having a size, a finite lifespan and constituting a certain proportion of allocations. Payloads also have a type and may be either arrays of certain primitives or instances of runtime-generate reflexive classes.
+
+Payloads are grouped into payload sets, each set having a start time, an end time and a data rate. The data rate is divided among any directly nested payloads according to their proportion fields and is inherited by any subsets. Subsets may override any parameters specified in parent sets. Payloads may inherit certain parameters from their parent set including their type.
+
+![Sessions workload design](doc/diagrams/Fig1_SyntheticWorkloadDiagram.png)
```

Comment:
I will look into why this isn't being displayed properly and upload the changes. (Same with the other broken links)

Reply:
Images and links work fine if you view the README when going through root if directory (rather than looking at preview through PR)